src/pointer-analysis/value_set_dereference.cpp

```diff
@@ -248,6 +215,45 @@ exprt value_set_dereferencet::dereference(
   return result_value;
 }
 
+value_set_dereferencet::valuet value_set_dereferencet::get_failure_value(
+  const exprt &pointer,
+  const typet &type)
+{
+  // first see if we have a "failed object" for this pointer
+  exprt failure_value;
+
+  if(
+    const symbolt *failed_symbol =
```

Comment:
I really, *really* do not like assignment statements inside the condition of if statements. I know you haven't introduced this, but could you perhaps clean it up? 🙏

Reply:
@NlightNFotis Actually, I really prefer this style so I wouldn’t consider that a cleanup.

Reply:
I'm with @NlightNFotis on this one, intentional assignments in conditions make it much harder to check for `=` vs. `==` bugs.

Reply:
This feels rather like the kind of thing that should be in the coding standard.